### ISSUE TITLE
global plopfile?

### ISSUE THREAD

**Issue by thejohnbackes** [amwmedia/plop#50]
how would you suggest creating a global plopfile that can be used across projects?

**Comment by amwmedia** [comment]
Pretty ironic timing on this because I'm writing this functionality into plop right now. In fact today I pushed up a feature branch to node-plop that implements part of it.

If you need something right away there is [an approach out there](https://github.com/amwmedia/plop/issues/20) where you can publish your plopfiles and templates up as an NPM module and just `require()` it into another project. There are several people and companies out there doing this exact thing but it has its downsides. Mainly you have to reference the plop templatesFiles using a path that starts with `node_modules/npm-module-name/` and there's no real isolation from the main plopfile. 

The new api I'm developing will be to use `plop.load()` to load assets from another plopfile or published npm module. It'll allow you to import JUST the generators, JUST the helpers, partials, or any combination. It will also allow you to provide a config object to the plopfile that can be used to alter its behavior.

When this is complete you should be able to do something like the following:
```javascript
plop.setGenerator('this-is-local', {blah blah blah});

// configObject would be passed to the plopfile when it's run
// this load would import just generators and helpers into the local plop context
plop.load('some-npm-installed-module', configObject, {generators: true, helpers: true});
```

Hope that helps 😃

**Comment by amwmedia** [comment]
this has been implemented in [v1.7.0](https://github.com/amwmedia/plop/releases/tag/v1.7.0)

**Comment by thejohnbackes** [comment]
thanks!
